src/Servers/Kestrel/Core/src/Internal/Certificates/ICertificateConfigLoader.cs

```diff
@@ -7,8 +7,6 @@ namespace Microsoft.AspNetCore.Server.Kestrel.Core.Internal.Certificates
 {
     internal interface ICertificateConfigLoader
     {
-        bool IsTestMock { get; }
-
-        X509Certificate2 LoadCertificate(CertificateConfig certInfo, string endpointName);
+        (X509Certificate2, X509Certificate2Collection) LoadCertificate(CertificateConfig certInfo, string endpointName);
```

Comment:
```suggestion
        SslStreamCertificateContext LoadCertificate(CertificateConfig certInfo, string endpointName);
```

Reply:
Feels bad

Reply:
Given this is an internal interface, I don't think the signature of it is a huge deal, but why does returning an SslStreamCertificateContext feel bad?

Reply:
It makes decisions about resolving the cert and its intermediates at the wrong layer. This builds the chain, potentially downloads intermediates and more. I'd prefer if that happened in a controlled place.